### ISSUE TITLE
Sporadic failures in PackageTests/NewBuild/CmdRun/Terminate/cabal.test.hs

### ISSUE THREAD

**Issue by robx** [haskell/cabal#8080]
**Describe the bug**

I see two recent master CI failures with `PackageTests/NewBuild/CmdRun/Terminate/cabal.test.hs`:
- https://github.com/haskell/cabal/runs/5778387580?check_suite_focus=true
- https://github.com/haskell/cabal/runs/5756929983?check_suite_focus=true

That test was added in #7921.

```
+ /Users/runner/work/cabal/cabal/dist-newstyle-validate-ghc-9.0.2/build/x86_64-osx/ghc-9.0.2/cabal-install-3.7.0.0/x/cabal/build/cabal/cabal v2-run exe
Up to date
terminated

stderr:
*** Exception:  expected: exiting
CallStack (from HasCallStack):
  assertOutputContains, called at PackageTests/NewBuild/CmdRun/Terminate/cabal.test.hs:34:3 in main:Main

*** unexpected failure for PackageTests/NewBuild/CmdRun/Terminate/cabal.test.hs
```

**Comment by robx** [comment]
We've merged a likely fix with #8081, let's keep this open for a bit to see whether the test still fails, and if it looks fine this can be closed.

**Comment by robx** [comment]
I've checked the last couple CI runs, looks like this isn't failing anymore (though a bit hard to tell for sure with other failures). Closing, but do feel free to reopen if it starts failing again!
